src/metabase/lib/js.cljs

```diff
@@ -1031,6 +1041,111 @@
+                                     (js->clj values)))
+
+(defn ^:export coordinate-filter-parts
+  "Destructures a coordinate filter clause created by [[coordinate-filter-clause]]. Returns `nil` if the clause does not
+  match the expected shape. Unlike regular numeric filters, coordinate filters do not support `:is-null` and
+  `:not-null`. There is also a special `:inside` operator that requires both latitude and longitude columns."
+  [a-query stage-number a-filter-clause]
+  (when-let [filter-parts (lib.core/coordinate-filter-parts a-query stage-number a-filter-clause)]
+    (let [{:keys [operator column longitude-column values]} filter-parts]
+      #js {:operator        (name operator)
+           :column          column
+           :longitudeColumn longitude-column
+           :values          (to-array (map clj->js values))})))
+
+(defn ^:export boolean-filter-clause
+  "Creates a boolean filter clause based on FE-friendly filter parts. It should be possible to destructure each created
+  expression with [[boolean-filter-parts]]."
+  [operator column values]
+  (lib.core/boolean-filter-clause (keyword operator)
+                                  column
+                                  (js->clj values)))
+
+(defn ^:export boolean-filter-parts
+  "Destructures a boolean filter clause created by [[boolean-filter-clause]]. Returns `nil` if the clause does not match
+  the expected shape."
+  [a-query stage-boolean a-filter-clause]
+  (when-let [filter-parts (lib.core/boolean-filter-parts a-query stage-boolean a-filter-clause)]
+    (let [{:keys [operator column values]} filter-parts]
+      #js {:operator (name operator)
+           :column   column
+           :values   (to-array (map clj->js values))})))
+
+(defn ^:export relative-date-filter-clause
+  "Creates a relative date filter clause based on FE-friendly filter parts. It should be possible to destructure each
+   created expression with [[relative-date-filter-parts]]."
+  [column value unit offset-value offset-unit options]
+  (lib.core/relative-date-filter-clause column
+                                        (if (string? value) (keyword value) value)
+                                        (keyword unit)
+                                        offset-value
+                                        (some-> offset-unit keyword)
+                                        (-> options js->clj (update-keys js-key->cljs-key))))
+
+(defn ^:export relative-date-filter-parts
+  "Destructures a relative date filter clause created by [[relative-date-filter-clause]]. Returns `nil` if the clause
+  does not match the expected shape."
+  [a-query stage-string a-filter-clause]
+  (when-let [filter-parts (lib.core/relative-date-filter-parts a-query stage-string a-filter-clause)]
+    (let [{:keys [column value unit offset-value offset-unit options]} filter-parts]
+      #js {:column      column
+           :value       (if (keyword? value) (name value) value)
```

Comment:
If `value` is always a keyword or string, it's fine to call `(name "foo") ;=> "foo"`. That won't work if the value is sometimes a number or something, though.

Reply:
`value` can be a `:current` or a number

Reply:
Ack.